src/Facebook/Authentication/OAuth2Client.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\Authentication;
+
+use Facebook\Facebook;
+use Facebook\FacebookApp;
+use Facebook\FacebookRequest;
+use Facebook\FacebookResponse;
+use Facebook\FacebookClient;
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+
+/**
+ * Class OAuth2Client
+ * @package Facebook
+ */
+class OAuth2Client
+{
+
+  /**
+   * @const string The base authorization URL.
+   */
+  const BASE_AUTHORIZATION_URL = 'https://www.facebook.com';
+
+  /**
+   * The FacebookApp entity.
+   *
+   * @var FacebookApp
+   */
+  protected $app;
+
+  /**
+   * The Facebook client.
+   *
+   * @var FacebookClient
+   */
+  protected $client;
+
+  /**
+   * The version of the Graph API to use.
+   *
+   * @var string
+   */
+  protected $graphVersion;
+
+  /**
+   * The last request sent to Graph.
+   *
+   * @var FacebookRequest|null
+   */
+  protected $lastRequest;
+
+  /**
+   * @param FacebookApp $app
+   * @param FacebookClient $client
+   * @param string $graphVersion The version of the Graph API to use.
+   */
+  public function __construct(FacebookApp $app, FacebookClient $client, $graphVersion)
+  {
+    $this->app = $app;
+    $this->client = $client;
+    $this->graphVersion = $graphVersion;
+  }
+
+  /**
+   * Returns the last FacebookRequest that was sent.
+   * Useful for debugging and testing.
+   *
+   * @return FacebookRequest|null
+   */
+  public function getLastRequest()
+  {
+    return $this->lastRequest;
+  }
+
+  /**
+   * Get the metadata associated with the access token.
+   *
+   * @param AccessToken|string $accessToken The access token to debug.
+   *
+   * @return AccessTokenMetadata
+   */
+  public function debugToken($accessToken)
+  {
+    $params = ['input_token' => (string) $accessToken];
```

Comment:
I'm not sure I understand what you mean. The `debugToken()` method is part of the public API:

```php
$oAuth2Client = $fb->getOAuth2Client();
$tokenMetadata = $oAuth2Client->debugToken('{access-token}');
```

A lot of times `{access-token}` will be an `AccessToken` entity that was returned from a `getAccessToken()` method or something. So this allows for either a string or `AccessToken` entity to be passed as an argument without the user having to type-cast the access token before passing it in.